server/jobs/startertemplate_job.go

```diff
@@ -0,0 +1,169 @@
+package jobs
+
+import (
+	"context"
+	"fmt"
+
+	"sync"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/plugin"
+	"github.com/mattermost/mattermost/server/public/pluginapi"
+	"github.com/mattermost/mattermost/server/public/pluginapi/cluster"
+	"github.com/wiggin77/merror"
+
+	"github.com/mattermost/mattermost-plugin-starter-template/server/config"
+	"github.com/mattermost/mattermost-plugin-starter-template/server/store/kvstore"
+)
+
+type StarterTemplateJobSettings struct {
+}
+
+func (s *StarterTemplateJobSettings) Clone() *StarterTemplateJobSettings {
+	return &StarterTemplateJobSettings{}
+}
+
+type StarterTemplateJob struct {
+	mux      sync.Mutex
+	settings *StarterTemplateJobSettings
+	job      *cluster.Job
+	runner   *runInstance
+
+	id      string
+	papi    plugin.API
+	client  *pluginapi.Client
+	kvstore kvstore.KVStore
+}
+
+func NewStarterTemplateJob(id string, api plugin.API, client *pluginapi.Client, kvstore kvstore.KVStore) (*StarterTemplateJob, error) {
+	return &StarterTemplateJob{
+		id:      id,
+		papi:    api,
+		client:  client,
+		kvstore: kvstore,
+	}, nil
+}
+
+func (j *StarterTemplateJob) GetID() string {
+	return j.id
+}
+
+// OnConfigurationChange is called by the job manager whenenver the plugin settings have changed.
+// It is suggested that you stop the current job (if any) and start a new job (if enabled) with new settings.
```

Comment:
Architecturally: why? If it's a long running job, odds are decent it's halfway done and I'm not sure of the semantics of interrupting it midway (especially if it persists back to the database or filestore). Should we just let it finish?